src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -645,14 +647,14 @@ public ListBackupsResult listBackups(ListBackupsRequest listBackupsRequest) {
     public RestoreTableFromBackupResult restoreTableFromBackup(
         RestoreTableFromBackupRequest restoreTableFromBackupRequest) {
         if (backupManager.isPresent()) {
-            //validate we have a backup for this tenant-table and we're under the same context
+            if (!getMtContext().getContextOpt().isPresent()) {
+                throw new MtBackupException("Cannot do restore of backup without tenant specifier", null);
+            }
+            //validate we have a backup for this tenant-table
             try {
-                TenantTableBackupMetadata tenantBackupMetadata = backupManager.get()
-                    .getTenantTableBackupFromArn(restoreTableFromBackupRequest.getBackupArn());
-                checkArgument(tenantBackupMetadata.getTenantId().equals(getMtContext().getContext()),
-                    "Current context does not match ARN context");
-            } catch (IllegalArgumentException e) {
-                throw new MtBackupException("Error restoring table, invalid input.", e);
+                describeBackup(new DescribeBackupRequest().withBackupArn(restoreTableFromBackupRequest.getBackupArn()));
```

Comment:
Why is it validating this explicitly up-front (as opposed to throwing an exception during restore when discovering a missing s3 file or something)?

Reply:
I wanna say it was an NPE when the backup ARN wasn't found, but that's not showing up when the ARN comes out of listBackups (vs ZOS duplicating ARN construction logic). Just removed this snippet